system/sched_note/note_main.c

```diff
@@ -83,6 +83,36 @@ static FAR const char *g_statenames[] =
  * Private Functions
  ************************************************************************************/
 
+/************************************************************************************
+ * Name: trace_dump_unflatten
+ ************************************************************************************/
+
+static void trace_dump_unflatten(FAR void *dst,
+                                 FAR uint8_t *src, size_t len)
+{
+  switch (len)
+    {
+#ifdef CONFIG_HAVE_LONG_LONG
+      case 8:
+        *(uint64_t *)dst = ((uint64_t)src[7] << 56)
+                         + ((uint64_t)src[6] << 48)
+                         + ((uint64_t)src[5] << 40)
+                         + ((uint64_t)src[4] << 32);
+#endif
+      case 4:
+        *(uint32_t *)dst = ((uint64_t)src[3] << 24)
+                         + ((uint64_t)src[2] << 16);
+      case 2:
+        *(uint16_t *)dst = ((uint64_t)src[1] << 8);
+      case 1:
+        *(uint8_t *)dst = src[0];
+        break;
```

Comment:
This code seems to be totally wrong.
We need to use temporary variable here and memcpy at the end:
```
  uint64_t tmp = 0;

  switch (len)
    {
#ifdef CONFIG_HAVE_LONG_LONG
      case 8:
        tmp += ((uint64_t)src[7] << 56);
        tmp += ((uint64_t)src[6] << 48);
        tmp += ((uint64_t)src[5] << 40);
        tmp += ((uint64_t)src[4] << 32);
#endif
      case 4:
        tmp += ((uint64_t)src[3] << 24);
        tmp += ((uint64_t)src[2] << 16);
      case 2:
        tmp += ((uint64_t)src[1] << 8);
      case 1:
        tmp += src[0];
        break;
      default:
        DEBUGASSERT(FALSE);
        break;
    }

  memcpy(dst, &tmp, len);
```